File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums.sort()
        ans=1
        if len(nums)==0:
            return 0
        c=1
        for i in range(len(nums)-1):
            if nums[i]+1==nums[i+1]:
                c+=1
                ans=max(ans,c)
            elif nums[i]==nums[i+1]:
                continue
            else:
                c=1
        return ans

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int res=1;
        int cur=1;
        if(nums.size()==0 || nums.size()==1){
            return nums.size();
        }
        set<int>s;
        for(auto i:nums){
            s.insert(i);
        }
        auto fast=s.begin();
        auto slow=s.begin();
        fast++;
        while (fast!=s.end()){
            if (*fast==*slow+1){
                cur+=1;
            }
            else{
                res=max(res,cur);
                cur=1;
            }
            fast++;
            slow++;
        }
        return max(cur,res);
    }
};